### PR TITLE
feat: discard endpoint

### PR DESCRIPTION
**Pull request by Marc-Roig**
### What does it do?

Adds discard endpoints:
`POST /collection-types/:model/:id/actions/discard`
`POST /single-types/:model/actions/discard`

### CODE REVIEW

**Comment by Marc-Roig** [comment]
I am working on the api tests in the meantime, but opening the PR to get a quicker review

**Comment by joshuaellis** [comment]
Is it possible to unpublish and discard at the same time in this PR or will that be a different PR? :)

**Comment by Marc-Roig** [comment]
@joshuaellis it's not in this PR but I don't mind adding it, do you we have a PR for that endpoint?

**Comment by Marc-Roig** [comment]
nvm! Found [Content-2038](https://strapi-inc.atlassian.net/browse/CONTENT-2038) where we specified the discardDraft  param on the unpublish endpoint, will create another PR on top of this to implement it :)

**Comment by joshuaellis** [comment]
I've QA'd discard on collection-types and it worked a treat ⭐

File: packages/core/content-manager/shared/contracts/single-types-v5.ts [hunk 1]
```diff
@@ -108,6 +108,23 @@ export declare namespace UnPublish {
   }
 }
 
+/**
+ * POST /single-types/:model/actions/discard
+ */
+export declare namespace Discard {
```
Comment (joshuaellis) [inline packages/core/content-manager/shared/contracts/single-types-v5.ts:114]: this needs Params

File: packages/core/content-manager/server/src/controllers/single-types.ts [hunk 2]
```diff
@@ -216,10 +216,47 @@ export default {
...
+    const documentMetadata = getService('document-metadata');
+    const permissionChecker = getService('permission-checker').create({ userAbility, model });
+
+    // Discarding updates the draft version, so it behaves like an update
+
+    if (permissionChecker.cannot.update()) {
+      return ctx.forbidden();
+    }
+
+    const sanitizedQuery = await permissionChecker.sanitizedQuery.unpublish(query);
```
Comment (joshuaellis) [inline packages/core/content-manager/server/src/controllers/single-types.ts:241]: should this be `.update` like collection-types?

Comment (Marc-Roig) [reply packages/core/content-manager/server/src/controllers/single-types.ts:241]: good catch, yes it should

**Review by jhoward1994 — APPROVED** [review]

**Pull request merged**